client/src/main/java/com/sanction/thunder/ThunderBuilder.java

```diff
@@ -24,38 +24,49 @@ public ThunderBuilder(String endpoint, String apiUser, String apiSecret) {
     Objects.requireNonNull(apiUser);
     Objects.requireNonNull(apiSecret);
 
-    ObjectMapper mapper = new ObjectMapper();
-
-    adapter = new RestAdapter.Builder()
-        .setEndpoint(endpoint)
-        .setConverter(new JacksonConverter(mapper))
-        .setRequestInterceptor(new ApiKeyInterceptor(apiUser, apiSecret))
-        .build();
+    retrofit = new Retrofit.Builder()
+      .baseUrl(endpoint)
+      .addConverterFactory(JacksonConverterFactory.create())
+      .client(buildHttpClient(apiUser, apiSecret))
+      .build();
   }
 
   /**
    * Build an instance of a ThunderClient.
    */
   public ThunderClient newThunderClient() {
-    return adapter.create(ThunderClient.class);
+    return retrofit.create(ThunderClient.class);
   }
 
-  private static final class ApiKeyInterceptor implements RequestInterceptor {
-    private final String authorization;
+  /**
+   * Create a new HttpClient that injects authorization into the client.
+   *
+   * @param user The API username to use when connecting to the endpoint.
+   * @param secret The API secret to use when connecting to the endpoint.
+   * @return The built OkHttpClient.
+   */
+  private OkHttpClient buildHttpClient(String user, String secret) {
+    Objects.requireNonNull(user);
+    Objects.requireNonNull(secret);
+
+    String token = Base64.getEncoder()
+        .encodeToString(String.format("%s:%s", user, secret).getBytes());
```

Comment:
Can you specify the charset when getting bytes? Java assumes UTF-8 but things can go wrong if the assumption is wrong: https://stackoverflow.com/questions/12659417/why-does-javas-string-getbytes-uses-iso-8859-1

Reply:
Nice catch, I'll fix this